truss-chains/truss_chains/framework.py

```diff
@@ -1211,133 +1212,195 @@ def decorator(cls: Type[ChainletT]) -> Type[ChainletT]:
-    """The context manager ensures that modules imported by the Model/Chain
-     are removed upon exit.
-
-    I.e. aiming at making the import idempotent for common usages, although there could
-    be additional side effects not accounted for by this implementation."""
-    module_name = module_path.stem  # Use the file's name as the module name
-    if not os.path.isfile(module_path):
-        raise ImportError(
-            f"`{module_path}` is not a file. You must point to a python file where "
-            f"the entrypoint Chainlet is defined."
+
+class ChainletImporter(_ABCImporter):
+    @classmethod
+    def no_entrypoint_error(self, module_path: pathlib.Path) -> ValueError:
+        return ValueError(
+            "No `target_name` was specified and no Chainlet in "
+            f"`{module_path}` was tagged with `@chains.mark_entrypoint`. Tag "
+            "one Chainlet or provide the Chainlet class name."
         )
 
-    import_error_msg = f"Could not import `{module_path}`. Check path."
-    spec = importlib.util.spec_from_file_location(module_name, module_path)
-    if not spec or not spec.loader:
-        raise ImportError(import_error_msg)
-
-    module = importlib.util.module_from_spec(spec)
-    module.__file__ = str(module_path)
-    # Since the framework depends on tracking the source files via `inspect` and this
-    # depends on the modules bein properly registered in `sys.modules`, we have to
-    # manually do this here (because importlib does not do it automatically). This
-    # registration has to stay at least until the push command has finished.
-    if module_name in sys.modules:
-        raise ImportError(
-            f"{import_error_msg} There is already a module in `sys.modules` "
-            f"with name `{module_name}`. Overwriting that value is unsafe. "
-            "Try renaming your source file."
+    @classmethod
+    def multiple_entrypoints_error(
+        self, module_path: pathlib.Path, entrypoints: set[type[definitions.ABCChainlet]]
+    ) -> ValueError:
+        return ValueError(
+            "`target_name` was not specified and multiple Chainlets in "
+            f"`{module_path}` were tagged with `@chains.mark_entrypoint`. Tag "
+            "one Chainlet or provide the Chainlet class name. Found Chainlets: "
+            f"\n{list(cls.name for cls in entrypoints)}"
         )
 
-    sys.modules[module_name] = module
-    # Add path for making absolute imports relative to the source_module's dir.
-    sys.path.insert(0, str(module_path.parent))
+    @classmethod
+    def target_cls_type(self) -> Type[definitions.ABCChainlet]:
+        return public_api.ChainletBase
 
-    return module, spec.loader
 
+class ModelImporter(_ABCImporter):
+    @classmethod
+    def no_entrypoint_error(self, module_path: pathlib.Path) -> ValueError:
+        return ValueError(
+            f"No Model class in `{module_path}` inherits from {self.target_cls_type()}."
```

Comment:
Previously, the error message would point to [ABCChainlet](https://github.com/basetenlabs/truss/blob/934bcbc2f7287b3f32fb2323ecf53e021c34598e/truss-chains/truss_chains/framework.py?plain=1#L1251), but users aren't interacting with that class directly. I think that including the public_api class is slightly better here